diff --git a/node_modules/cypress-split/src/index.js b/node_modules/cypress-split/src/index.js
index 7ec3103..9820b6e 100644
--- a/node_modules/cypress-split/src/index.js
+++ b/node_modules/cypress-split/src/index.js
@@ -89,6 +89,8 @@ function cypressSplit(on, config) {
   let SPLIT = process.env.SPLIT || config.env.split || config.env.SPLIT
   let SPLIT_INDEX = process.env.SPLIT_INDEX || config.env.splitIndex
   let SPLIT_FILE = process.env.SPLIT_FILE || config.env.splitFile
+  let OUTPUT_FILE = process.env.OUTPUT_FILE || config.env.outputFile || SPLIT_FILE
+
 
   // some CI systems like TeamCity provide agent index starting with 1
   // let's check for SPLIT_INDEX1 and if it is set,
@@ -115,12 +117,7 @@ function cypressSplit(on, config) {
         // make sure every spec filename is absolute
         return path.resolve(specFilename)
       })
-    console.log(
-      '%s have explicit %d spec %s',
-      label,
-      specs.length,
-      specs.length === 1 ? 'file' : 'files',
-    )
+    
   }
 
   if (SPLIT === 'true' || SPLIT === true) {
@@ -129,12 +126,7 @@ function cypressSplit(on, config) {
     if (process.env.CIRCLECI) {
       SPLIT = process.env.CIRCLE_NODE_TOTAL
       SPLIT_INDEX = process.env.CIRCLE_NODE_INDEX
-      console.log(
-        '%s detected CircleCI machine %d of %d',
-        label,
-        SPLIT_INDEX,
-        SPLIT,
-      )
+
     } else if (process.env.CI_NODE_INDEX && process.env.CI_NODE_TOTAL) {
       // GitLab CI
       // https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
@@ -142,12 +134,7 @@ function cypressSplit(on, config) {
       // GitLabCI index starts with 1
       // convert it to zero base
       SPLIT_INDEX = Number(process.env.CI_NODE_INDEX) - 1
-      console.log(
-        '%s detected GitLabCI machine %d of %d',
-        label,
-        SPLIT_INDEX,
-        SPLIT,
-      )
+
     } else {
       throw new Error('Do not know how to determine the correct split')
     }
@@ -318,8 +305,8 @@ function cypressSplit(on, config) {
         console.log(timingsString)
 
         if (!foundSplitFile) {
-          console.log('%s writing out timings file %s', label, SPLIT_FILE)
-          fs.writeFileSync(SPLIT_FILE, timingsString + '\n', 'utf8')
+          console.log('%s writing out timings file %s', label, OUTPUT_FILE)
+          fs.writeFileSync(OUTPUT_FILE, timingsString + '\n', 'utf8')
         } else {
           const splitFile = JSON.parse(fs.readFileSync(foundSplitFile, 'utf8'))
           const hasUpdatedTimings = hasTimeDifferences(splitFile, timings, 0.1)
@@ -331,18 +318,22 @@ function cypressSplit(on, config) {
             console.log(
               '%s writing out updated timings file %s',
               label,
-              SPLIT_FILE,
+              OUTPUT_FILE,
             )
             debug('previous timings has %d entries', splitFile.durations.length)
             debug('current timings has %d entries', timings.durations.length)
             debug(
               'merged timings has %d entries written to %s',
               mergedTimings.durations.length,
-              foundSplitFile,
+              OUTPUT_FILE,
             )
-            fs.writeFileSync(foundSplitFile, mergedText + '\n', 'utf8')
+            fs.writeFileSync(OUTPUT_FILE, mergedText + '\n', 'utf8')
           } else {
             console.log('%s spec timings unchanged', label)
+            if(OUTPUT_FILE !== SPLIT_FILE){
+              console.log('%s writing out timings file %s', label, OUTPUT_FILE)
+              fs.writeFileSync(OUTPUT_FILE, timingsString + '\n', 'utf8')
+            }
           }
         }
       }
